src/components/SearchResultDrawer/index.tsx

```diff
@@ -0,0 +1,272 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState
+} from 'react';
+
+import {
+  DrawerProps
+} from 'antd';
+import { ColumnType } from 'antd/es/table';
+
+import OlFeature from 'ol/Feature';
+import GeoJSONParser from 'ol/format/GeoJSON';
+import OlLayerVector from 'ol/layer/Vector';
+import OlSourceVector from 'ol/source/Vector';
+import OlStyleCircle from 'ol/style/Circle';
+import OlStyleFill from 'ol/style/Fill';
+import OlStyleStroke from 'ol/style/Stroke';
+import OlStyle from 'ol/style/Style';
+
+import {
+  useTranslation
+} from 'react-i18next';
+
+import PropertyGrid from '@terrestris/react-geo/dist/Grid/PropertyGrid/PropertyGrid';
+import { useMap } from '@terrestris/react-util/dist/Hooks/useMap/useMap';
+
+import {
+  PropertyFormItemReadConfig, PropertyFormTabConfig
+} from '@terrestris/shogun-util/dist/model/Layer';
+
+import useAppDispatch from '../../hooks/useAppDispatch';
+import useAppSelector from '../../hooks/useAppSelector';
+import {
+  setSearchResultState
+} from '../../store/searchResult';
+import MapDrawer from '../MapDrawer';
+
+export type SearchResultDrawerProps = DrawerProps;
+
+type AttributeRecord = {
+  key: string;
+  attributeName: string;
+  attributeValue: string | number | string[] | number[];
+};
+
+export const SearchResultDrawer: React.FC<SearchResultDrawerProps> = ({
+  ...passThroughProps
+}): JSX.Element => {
+
+  const dispatch = useAppDispatch();
+  const map = useMap();
+  const isSearchResultDrawerVisible = useAppSelector(state => state.searchResult.drawerVisibility);
+  const geoJSONFeature = useAppSelector(state => state.searchResult.geoJSONFeature);
+  const [highlightLayer, setHighlightLayer] = useState<OlLayerVector<OlSourceVector> | null>(null);
+  const olFeature: OlFeature | null = useMemo(() => {
+    if (!geoJSONFeature) {
+      return null;
+    }
+    return new GeoJSONParser().readFeature(geoJSONFeature) as OlFeature;
+  }, [geoJSONFeature]);
+
+  useEffect(() => {
+    if (!map) {
+      return;
+    }
+
+    const layer = new OlLayerVector({
+      source: new OlSourceVector(),
+      style: new OlStyle({
+        stroke: new OlStyleStroke({
+          color: 'rgb(255,0,0)',
+          width: 2
+        }),
+        fill: new OlStyleFill({
+          color: 'rgba(255,255,255, 0.5)'
+        }),
+        image: new OlStyleCircle({
+          radius: 10,
+          fill: new OlStyleFill({
+            color: 'rgba(255,255,255, 0.5)'
+          }),
+          stroke: new OlStyleStroke({
+            color: 'rgb(255,0,0)',
+            width: 3
+          })
+        })
+      })
+    });
+
+    setHighlightLayer(layer);
+    map.addLayer(layer);
```

Comment:
Do we need to remove the layer on unmount? We also have [this](https://github.com/terrestris/react-util/blob/main/src/Hooks/useOlLayer/useOlLayer.ts) hook for comparable scenarios.

Reply:
Yes, the hook from react-util will be used here now.